src/uct/api/uct_def.h

```diff
@@ -235,4 +238,59 @@ typedef size_t (*uct_pack_callback_t)(void *dest, void *arg);
 typedef void (*uct_unpack_callback_t)(void *arg, const void *data, size_t length);
 
 
+/**
+ * Callback to handle unexpected eager tag.
```

Comment:
1. @brief is missing. 
2. please describe constrains and requirements associated with the callback. Does it have to be thread or signal safe ? Can the callback call other communication routines ?